src/Containers.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import Listing from '../lib/cockpit-components-listing.jsx';
+import cockpit from 'cockpit';
+
+import ContainerDetails from './ContainerDetails.js';
+const _ = cockpit.gettext;
+
+class Containers extends React.Component {
+    constructor(props) {
+	super(props);
+	console.log(props.containers);
+	this.renderRow = this.renderRow.bind(this);
+	this.getContainerState = this.getContainerState.bind(this);
+    }
+
+    navigateToContainer (container) {
+	cockpit.location.go([ container.ID ]);
+    }
+    //TODO
+    deleteContainer(event, container) {
+	return undefined;
+    }
+    //TODO
+    stopContainer(container) {
+	return undefined;
+    }
+    //TODO
+    startContainer(container) {
+	return undefined;
+    }
+
+    //TODO
+    restartContainer(container) {
+	return undefined;
+    }
+
+    getContainerState(state) {
+	if (state.Status)
+	    return state.Status;
+	if (state.Running)
+	    return "running";
+	if (state.Paused)
+	    return "paused";
+	if (state.Restarting)
+	    return "restarting";
+	if (state.FinishedAt && state.FinishedAt.indexOf("0001") === 0)
+	    return "created";
+	return "exited";
+    }
+    renderRow(container) {
+	let isRunning = !!container.State.Running;
+	//TODO: check container.Image == container.ImageID
+	let image = container.ImageName;
+	let state = this.getContainerState(container.State);
+
+	let columns = [
+	    { name: container.Name.replace(/^\//, ''), header: true },
+	    image,
+	    // util.render_container_cmdline(container),
+	    container.Config.Entrypoint === "" ? container.Config.Cmd.join(" "):container.Config.Cmd.join(""),
```

Comment:
same value in both cases, and missing spaces around operators. You probably should enable eslint for this project. (And I'll look into doing that for the starter kit)

Reply:
Not the same. Add a space in the first case. The other is to join the directory.